e2e/utils/test_cases.js

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+// See reference: https://support.smartbear.com/tm4j-cloud/api-docs/
+
+const axios = require('axios');
+
+const {getAllTests} = require('./report');
+
+const status = {
+    passed: 'Pass',
+    failed: 'Fail',
+    pending: 'Pending',
+    skipped: 'Skip',
+};
+
+const environment = {
+    chrome: 'Chrome',
+    firefox: 'Firefox',
+};
+
+function getStepStateResult(steps = []) {
+    return steps.reduce((acc, item) => {
+        if (acc[item.state]) {
+            acc[item.state] += 1;
+        } else {
+            acc[item.state] = 1;
+        }
+
+        return acc;
+    }, {});
+}
+
+function getStepStateSummary(steps = []) {
+    const result = getStepStateResult(steps);
+
+    return Object.entries(result).map(([key, value]) => `${value} ${key}`).join(',');
+}
+
+function getTM4JTestCases(report) {
+    return getAllTests(report.results).
+        filter((item) => /^(MM-T)\w+/g.test(item.title)).
```

Comment:
For some reason, the regex didn't work as expected on some cases when I assigned it initially in a variable. Directly using it did the trick.

Reply:
I think you need to use `new RegEx(/^(MM-T)\w+/g)` for variable assignment

Reply:
That's interesting and so I tried. However, same result, it didn't work. When assigned to a variable, all patterns from this [json](https://mm-cypress-report.s3.amazonaws.com/703-integrate-tm4j-master/all.json) were matched except with `MM-T1784_1` up to `MM-T1784_4` which got 2 of 4 steps only. But with `/^(MM-T)\w+/g.test(item.title)`, all 4 got the match.